src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -123,37 +122,43 @@ def test_fusetrans_error_not_same_parent():
 
 
 # ----------------------------------------------------------------------------
-def fuse_loops(parser=None, fortran_code=None):
+def fuse_loops(parser, fortran_code):
     '''Helper function that fuses the first two nodes in the given
     Fortran code, and returns the fused Fortran code as string.
+    If an error is detected by the used NemoLoopFuseTrans transformation,
+    it will raise a TransformationError.
+
+    :param parser: an fparser2 program class.
+    :type parser: :py:class:`fparser.two.Fortran2003.Program`
 
-    :param string fortran_code: the Fortran code to loop fuse.
+    :param str fortran_code: the Fortran code to loop fuse.
+    :param parser:
 
-    :returns: String representation of fused code.
-    :rtype: str
+    :returns: a 2-tuple of the fused Fortran code, and the PSyIR \
+        representation of the supplied Fortran code.
+    :rtype: 2-tuple of (str, :py:class:`psyclone.psyir.nodes.Container` or \
+                             :py:class:`psyclone.psyir.nodes.Routine`)
 
-    :raises TransformationError: if the loop fuse transformation cannot
-        be applied.
     '''
     reader = FortranStringReader(fortran_code)
-    ast = parser(reader)
-    psy = PSyFactory("nemo").create(ast)
-    schedule = psy.invokes.get("sub").schedule
-
-    loop1 = schedule.children[0]
-    loop2 = schedule.children[1]
+    fp2_ast = parser(reader)
+    fp2_reader = Fparser2Reader()
+    psyir = fp2_reader.generate_psyir(fp2_ast)
+    loop1 = psyir.children[0]
+    loop2 = psyir.children[1]
```

Comment:
This touches on what I'm doing in #1210 - eventually you'll apply a Transformation to the generic PSyIR and it will give you NEMO-Specific PSyIR. In fact, please could you add a TODO here and reference that issue.

Reply:
Done.